client/src/app/shared/directives/repo-observable.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { Directive, Input, OnInit } from '@angular/core';
+
+import { ModelRequestRepository } from 'app/core/repositories/model-request-repository';
+import { ComponentServiceCollector } from 'app/core/ui-services/component-service-collector';
+import { BaseModelContextComponent } from 'app/site/base/components/base-model-context.component';
+
+/**
+ * A simply directive to request all models of one repository.
+ *
+ * Useful to watch all models although a component does not effectively request them.
+ */
+@Directive({
+    selector: '[osRepoObservable]'
+})
+export class RepoObservableDirective extends BaseModelContextComponent implements OnInit {
+    @Input()
+    public osRepoObservable: ModelRequestRepository;
+
+    public constructor(componentServiceCollector: ComponentServiceCollector) {
+        super(componentServiceCollector);
+    }
+
+    public ngOnInit(): void {
+        if (!this.osRepoObservable) {
+            return;
+        }
+        this.doModelRequest();
+    }
+
+    private doModelRequest(): void {
+        this.requestModels(this.osRepoObservable.getRequestToGetAllModels());
+    }
```

Comment:
This seems dangerously if used in for loops what so ever. Also a `ngOnDestroy` is missing. this would leak tcp connections! Make sure the usecase is right - Requesting models is a very heavy task, which also must be cleaned up carefully.

Reply:
Ok, nvm. See other comment about removing this directive.